static/packages/autocomplete.js

```diff
@@ -17,124 +17,97 @@ class AutocompleteField {
         const place = self.autocomplete.getPlace();
         let address = new Address(place, self.input.value);
 
-        self.input.value = address.fields['formatted'];
-        self.city.value = address.fields['city'];
-        self.street.value = address.fields['street'];
-        self.streetNumber.value = address.fields['street_number'];
+        self.input.value = address.formatted;
+        self.city.value = address.city;
+        self.street.value = address.street;
+        self.streetNumber.value = address.street_number;
     }
 }
 
 class Address {
     constructor(place, input) {
         if (typeof place === 'undefined' || typeof place.address_components === 'undefined')
-            this.fields = this.customAddressFormatting(this, input);
+            this.customAddressFormatting(this, input);
         else
-            this.fields = this.extractAddressComponents(place);
+            this.extractAddressComponents(this, place);
     }
 
     customAddressFormatting(self, customInput) {
+        const numRegex = /\d+/;
+
         let components = customInput.split(',');
-        let address = {}
+        let streetNumber;
+        let street;
+        let city;
 
-        if (components.length === 1)
-            address['city'] = components[0];
+        if (components.length == 1)
+            city = components[0];
         else {
             if (components[1].charAt(0) === ' ')
                 components[1] = components[1].substring(1);
 
-            address['city'] = components[1];
-            streetDetails = self.extractStreetDetails(components[0].split(' '));
-
-            address['street'] = streetDetails[0];
-            address['street_number'] = streetDetails[1];
-        }
-
-        if (typeof address['street'] === 'undefined')
-            address['street'] = '<no street>';
-        if (typeof address['street_number'] === 'undefined' || isNaN(address['street_number']))
-            address['street_number'] = 1;
-
-        address['formatted'] = customInput;
-
-        return address;
-    }
-
-    extractStreetDetails(self, detailsStr) {
-        let word = "";
-        let streetName = "";
-        let streetNumber;
-
-        streetDetails = detailsStr;
+            if (streetNumber = + components[0].match(numRegex)) {
+                street = components[0];
+                city = components[1];
+            }
+            else {
+                streetNumber = +components[1].match(numRegex);
+                street = components[1];
+                city = components[0];
+            }
 
-        for (i = 0; !self.isNumber(word); i++) {
-            word = streetDetails[i];
-            streetName += streetDetails + ' ';
+            street = street.replace(/[0-9]/g, '').slice(0, -1);
+            alert("'" + String(street) + "'");

```

Comment:
Yeah.. no

Reply:
yup.. uncomitted changes

Reply:
wait no, Did you pull it? I did remove it on this branch